guava/src/main/java/com/fasterxml/jackson/datatype/guava/ser/RangeSerializer.java

```diff
@@ -72,7 +90,7 @@ public JsonSerializer<?> createContextual(SerializerProvider prov,
             // let's not consider "untyped" (java.lang.Object) to be meaningful here...
             if (endpointType != null && !endpointType.hasRawClass(Object.class)) {
```

Comment:
Maybe also check whether `shape` has changed; no need to create new instance if it hasn't?